saleor/core/notification/validation.py

```diff
@@ -0,0 +1,65 @@
+from django.core.exceptions import ValidationError
+
+from ...account.models import User
+from ...account.notifications import get_user_custom_payload
+from ...core.permissions import AccountPermissions, OrderPermissions
+from ...graphql.core.enums import ExternalNotificationTriggerErrorCode
+from ...graphql.utils import resolve_global_ids_to_primary_keys
+from ...order.models import Order
+from ...order.notifications import get_custom_order_payload
+
+PAYLOAD_MAPPING_FOR_CUSTOM_NOTIFICATION = {
+    "User": (
+        User,
+        get_user_custom_payload,
+        AccountPermissions.MANAGE_USERS,
+    ),
+    "Order": (
+        Order,
+        get_custom_order_payload,
+        OrderPermissions.MANAGE_ORDERS,
+    ),
+}
+
+
+def validate_ids_and_get_model_type_and_pks(data_input):
+    if ids := data_input.get("ids"):
+        model_type, pks = resolve_global_ids_to_primary_keys(ids)
+        return model_type, pks
+    raise ValidationError(
+        {
+            "ids": ValidationError(
+                "The obligatory param 'ids' is missing or is empty.",
+                code=ExternalNotificationTriggerErrorCode.REQUIRED,
+            )
+        }
+    )
+
+
+def validate_and_get_external_event_type(data_input):
+    if external_event_type := data_input.get("external_event_type"):
+        return external_event_type
+    raise ValidationError(
+        {
+            "external_event_type": ValidationError(
+                "The obligatory param 'external_event_type' is missing or is empty.",
+                code=ExternalNotificationTriggerErrorCode.REQUIRED,
+            )
+        }
+    )
+
+
+def validate_and_get_payload_params(model_type):
+    if payload_params := PAYLOAD_MAPPING_FOR_CUSTOM_NOTIFICATION.get(model_type):
+        return payload_params
+    raise ValidationError(
+        {
+            "external_notification_trigger": ValidationError(
+                f"Wrong type of model."
+                f"ExternalNotificationTrigger is suitable"
+                f"for {','.join(PAYLOAD_MAPPING_FOR_CUSTOM_NOTIFICATION.keys())} "
+                f"models.",
```

Comment:
Can't we have instead:
```suggestion
                f"Wrong type of model - ExternalNotificationTrigger is suitable for "
                f"{','.join(PAYLOAD_MAPPING_FOR_CUSTOM_NOTIFICATION.keys())} models.",
```
Seems a bit simpler.

Reply:
Flake8 will throw `saleor/core/notification/validation.py:60:89: E501 line too long (90 > 88 characters)` if i back to previous version.

Reply:
You sure? If I paste that to my editor it seems under 88 chars:
![image](https://user-images.githubusercontent.com/5421321/130611088-6c8ac6c6-a3b3-49fd-b635-91cb5e0fab70.png)

We should at least fix the spaces, now this part:
```
f"Wrong type of model."
f"ExternalNotificationTrigger is suitable"
```
will be printed as:
```
Wrong type of model.ExternalNotificationTrigger is suitable
```

Reply:
Is this an error returned by GraphQL? If so, there are no "models" in GraphQL, it has "object types."